### ISSUE TITLE
Support for adding ARIA label to Numeral Date Field without displaying visible labels.

### ISSUE THREAD

**Issue by siddhant-sage24** [Sage/carbon#7146]
### Description

As a developer using Carbon, I want the Numeral Date Field component to support ARIA labels (such as aria-label or ariaLabe) without requiring a visible label, so that the component remains accessible for screen readers.

### Suggested solution

Please can you add support for an aria-label/ariaLabel or similar prop in the Numeral Date Field component, allowing it to provide descriptive labels for screen readers without rendering visible labels.

### Confidentiality

- [X] I confirm there is no confidential or commercially sensitive information included.

**Comment by sianford** [comment]
@ljemmo @harpalsingh can you please have a look at this and give your opinion on whether this should be an allowed state? It would result in the Numeral Date Input not having a visible label. This has only recently been brought in line with the DS.

**Comment by siddhant-sage24** [comment]
> I can think of some edge cases where this might be needed but its not actively something we would encourage for accessibility reasons. @siddhant-sage24 - do you have an example of the design you can share with us for more context. Feel free to DM me on slack.

Okay, I will connect with you on Slack @ljemmo

**Comment by edleeks87** [comment]
Hey @ljemmo did we get anywhere with this request?

**Comment by siddhant-sage24** [comment]
Hi @edleeks87 @ljemmo , I asked on accessibility channel about this, also checked for work around. But its not giving any solution for this requirement. We will need a "aria-label" prop added to Numeral-Date component, similar to other carbon components in our Carbon library.

**Comment by siddhant-sage24** [comment]
> FE-7054 has been raised, my suggestion here is for Carbon to automatically provide an aria-labelledby link to the associated label above each field. i.e (day, month, year), this will also then work with the translations. Is this acceptable [@siddhant-sage24](https://github.com/siddhant-sage24)?

Thanks. Yes, this will work.

<img width="1000" alt="Image" src="https://github.com/user-attachments/assets/e54046f0-5675-4c47-8ea0-a5b76add776a" />

For example: Similar to these above "aria" props for other carbon components. If Carbon library is able to add it to "Numeral Date" field, it will be great.

**Comment by ljemmo** [comment]
@nicktitchmarsh   and @sianford  there's a reply from Chris here about use of aria: https://sageone.slack.com/archives/C01BE6N6XNK/p1736435262856349?thread_ts=1736432834.702219&cid=C01BE6N6XNK

Does this help unblock this issue?

**Comment by nicktitchmarsh** [comment]
Ok I'll discuss this solution with the team during our refinement

**Comment by nicktitchmarsh** [comment]
We actually already render this as a fieldset with individual fields and labels. I think the ask here is that @siddhant-sage24 wants to remove the labels of Day Month and Year in favour of aria-labels. I think we need to push back on this design requirement as it would be considered less accessible without labels. @ljemmo can you advise the designer of this screen?

**Comment by ljemmo** [comment]
@nicktitchmarsh if that is indeed the ask, then i agree that we shouldn't be doing that for accessibility reasons. Unsure who the designer is that i would need to contact here.

**Comment by nicktitchmarsh** [comment]
I spoke with Siddhant, it seems the issue here is that when the field is set to required, screen readers will read out the field name with "asterisk" as well. I have asked accessibility for a steer on how this should work

**Comment by nuria1110** [comment]
From the accessibility team:
"The DS uses the * to show a field is required, alongside a form key (Fill in all fields marked with *). Most screen readers will read out the star and screen reader users will understand the use of the star alongside the form key. The `required` attribute on the form fields is used as a fallback for screen readers that don't read the star. Passing an `aria-label` to show required field is unnecessary given the * and required attribute." 

Closing this ticket and the GH issue as this does not require a fix.
